packages/astro/src/@types/astro.ts

```diff
@@ -681,6 +682,7 @@ export interface AstroIntegration {
 		'astro:server:start'?: (options: { address: AddressInfo }) => void | Promise<void>;
 		'astro:server:done'?: () => void | Promise<void>;
 		'astro:build:start'?: (options: { buildConfig: BuildConfig }) => void | Promise<void>;
+		'astro:build:server:setup'?: (options: { vite: ViteConfigWithSSR }) => void;
```

Comment:
nit: can this be `astro:build:setup` to match the others? I'd really like to keep all of these to 3 words, max. 

Since this appears to run on every build (not just SSR) I think it makes more sense as well.

Reply:
Changed and added `target: 'server' | 'client'`